src/inspector_agent.cc

```diff
@@ -833,7 +833,8 @@ bool Agent::IsActive() {
 
 void Agent::WaitForConnect() {
   CHECK_NOT_NULL(client_);
-  client_->waitForFrontend();
+  if (!client_->hasConnectedSessions())
```

Comment:
I am not sure this change is needed. If I understand correctly, the goal here is to wait for a specific frontend to connect. This check will subvert this expectation.